### PR TITLE
Check that CSRFGuard is enabled before injecting tokens with JSP Tags…

### PR DESCRIPTION
**Pull request by alexhatz05**
Check that CSRFGuard is enabled before injecting tokens with JSP Tags [TokenTag, TokenNameTag, TokenValueTag]

### CODE REVIEW

File: csrfguard-extensions/csrfguard-jsp-tags/src/main/java/org/owasp/csrfguard/tag/TokenNameTag.java [hunk 1]
```diff
@@ -40,14 +40,16 @@
...
 	@Override
 	public int doStartTag() {
-		final String tokenName = CsrfGuard.getInstance().getTokenName();
-
-		try {
-			this.pageContext.getOut().write(tokenName);
-		} catch (final IOException e) {
-			this.pageContext.getServletContext().log(e.getLocalizedMessage(), e);
+		final CsrfGuard csrfGuard = CsrfGuard.getInstance();
+		final String tokenName = csrfGuard.getTokenName();
```
Comment (forgedhallpass) [inline csrfguard-extensions/csrfguard-jsp-tags/src/main/java/org/owasp/csrfguard/tag/TokenNameTag.java:44]: The `tokeName` does not have to be retrieved if CSRFGuard is disabled.

File: csrfguard-extensions/csrfguard-jsp-tags/src/main/java/org/owasp/csrfguard/tag/TokenNameTag.java [hunk 2]
```diff
@@ -40,14 +40,16 @@
...
+
+		if(csrfGuard.isEnabled()) {
```
Comment (forgedhallpass) [inline csrfguard-extensions/csrfguard-jsp-tags/src/main/java/org/owasp/csrfguard/tag/TokenNameTag.java:46]: please make sure to follow the same formatting as the rest of the code, and put a space after the `if` keyword.

File: csrfguard-extensions/csrfguard-jsp-tags/src/main/java/org/owasp/csrfguard/tag/TokenTag.java [hunk 3]
```diff
@@ -48,16 +48,16 @@ public int doStartTag() {
...
 		}
-
-		final LogicalSession logicalSession = csrfGuard.getLogicalSessionExtractor().extract((HttpServletRequest) this.pageContext.getRequest());
-		final String tokenValue = Objects.nonNull(logicalSession) ? csrfGuard.getTokenService().getTokenValue(logicalSession.getKey(), getUri()) : null;
-
-		try {
-			this.pageContext.getOut().write(tokenName + '=' + tokenValue);
-		} catch (final IOException e) {
-			this.pageContext.getServletContext().log(e.getLocalizedMessage(), e);
+		else if(csrfGuard.isEnabled()) {
```
Comment (forgedhallpass) [inline csrfguard-extensions/csrfguard-jsp-tags/src/main/java/org/owasp/csrfguard/tag/TokenTag.java:51]: Checking if CSRFGuard is the enabled should be the first condition.

Comment (forgedhallpass) [reply csrfguard-extensions/csrfguard-jsp-tags/src/main/java/org/owasp/csrfguard/tag/TokenTag.java:51]: Please put a space after the `if`

File: csrfguard-extensions/csrfguard-jsp-tags/src/main/java/org/owasp/csrfguard/tag/TokenValueTag.java [hunk 4]
```diff
@@ -47,16 +47,16 @@ public int doStartTag() {
...
 		}
-
-		final LogicalSession logicalSession = csrfGuard.getLogicalSessionExtractor().extract((HttpServletRequest) this.pageContext.getRequest());
-		final String tokenValue = Objects.nonNull(logicalSession) ? csrfGuard.getTokenService().getTokenValue(logicalSession.getKey(), getUri()) : null;
-
-		try {
-			this.pageContext.getOut().write(tokenValue);
-		} catch (final IOException e) {
-			this.pageContext.getServletContext().log(e.getLocalizedMessage(), e);
+		else if (csrfGuard.isEnabled()) {
```
Comment (forgedhallpass) [inline csrfguard-extensions/csrfguard-jsp-tags/src/main/java/org/owasp/csrfguard/tag/TokenValueTag.java:50]: same here

**Review by forgedhallpass — CHANGES_REQUESTED** [review]
Thank you for your contribution. Please fix the minor changes I've requested, then I'll merge it

**Review by alexhatz05 — COMMENTED** [review]
Implemented reviewer's comments.

File: csrfguard-extensions/csrfguard-jsp-tags/src/main/java/org/owasp/csrfguard/tag/TokenValueTag.java [hunk 5]
```diff
@@ -44,19 +44,19 @@
...
 		}
-
-		final LogicalSession logicalSession = csrfGuard.getLogicalSessionExtractor().extract((HttpServletRequest) this.pageContext.getRequest());
-		final String tokenValue = Objects.nonNull(logicalSession) ? csrfGuard.getTokenService().getTokenValue(logicalSession.getKey(), getUri()) : null;
-
-		try {
-			this.pageContext.getOut().write(tokenValue);
-		} catch (final IOException e) {
-			this.pageContext.getServletContext().log(e.getLocalizedMessage(), e);
+		else if (csrfGuard.isTokenPerPageEnabled() && Objects.isNull(getUri())) {
```
Comment (forgedhallpass) [inline csrfguard-extensions/csrfguard-jsp-tags/src/main/java/org/owasp/csrfguard/tag/TokenValueTag.java:57]: This condition should be within the `if` statement above, executed when the solution is enabled.

File: csrfguard-extensions/csrfguard-jsp-tags/src/main/java/org/owasp/csrfguard/tag/TokenTag.java [hunk 6]
```diff
@@ -45,19 +45,19 @@ public int doStartTag() {
...
 		}
-
-		final LogicalSession logicalSession = csrfGuard.getLogicalSessionExtractor().extract((HttpServletRequest) this.pageContext.getRequest());
-		final String tokenValue = Objects.nonNull(logicalSession) ? csrfGuard.getTokenService().getTokenValue(logicalSession.getKey(), getUri()) : null;
-
-		try {
-			this.pageContext.getOut().write(tokenName + '=' + tokenValue);
-		} catch (final IOException e) {
-			this.pageContext.getServletContext().log(e.getLocalizedMessage(), e);
+		else if (csrfGuard.isTokenPerPageEnabled() && Objects.isNull(getUri())) {
```
Comment (forgedhallpass) [inline csrfguard-extensions/csrfguard-jsp-tags/src/main/java/org/owasp/csrfguard/tag/TokenTag.java:58]: same here

File: csrfguard-extensions/csrfguard-jsp-tags/src/main/java/org/owasp/csrfguard/tag/TokenTag.java [hunk 7]
```diff
@@ -45,19 +45,19 @@ public int doStartTag() {
 		final CsrfGuard csrfGuard = CsrfGuard.getInstance();
 		final String tokenName = csrfGuard.getTokenName();
 
-		if (csrfGuard.isTokenPerPageEnabled() && Objects.isNull(getUri())) {
-			throw new IllegalStateException("Must define 'uri' attribute when token per page is enabled");
+		if (csrfGuard.isEnabled()) {
+			final LogicalSession logicalSession = csrfGuard.getLogicalSessionExtractor().extract((HttpServletRequest) this.pageContext.getRequest());
+			final String tokenValue = Objects.nonNull(logicalSession) ? csrfGuard.getTokenService().getTokenValue(logicalSession.getKey(), getUri()) : null;
```
Comment (forgedhallpass) [inline csrfguard-extensions/csrfguard-jsp-tags/src/main/java/org/owasp/csrfguard/tag/TokenTag.java:50]: the `tokenName` is enough to be retrieved if the guard is enabled.

**Review by forgedhallpass — CHANGES_REQUESTED** [review]
Please make sure to implement the changes, because the current logic contains some logical errors.

**Review by alexhatz05 — COMMENTED** [review]
Pushed another commit! Hope it's ok now :) Thanks for your support.

**Review by forgedhallpass — APPROVED** [review]

**Pull request merged**